tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+///\file
+/// Helper functions for testing coordinate maps
+
+#pragma once
+
+#include <array>
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
```

Comment:
Don't need anon namespace for function templates, only functions

Reply:
This is generally not true.  Function templates actually collide in a worse manner than functions do: it may silently do the wrong thing instead of generating a linking error like the function case.

Here, though, there shouldn't be an anonymous namespace because anonymous namespaces shouldn't be used in header files.

Reply:
function templates do not have multiple definition problems, the linker is required to sort those out because compilers will greedily generate instantiations in each translation unit. The reason we've been using anonymous namespaces has been to avoid multiple definitions that the linker can't sort out. The linker may choose the wrong definition, or one may unintentionally choose the wrong overloaded function template, but those are different issues. What are you referring to?

Reply:
Yes, the fact that the linker sorts them out is the problem (this may be the "wrong definition" you referred to above).  It will generally choose one of the instantiations to use and silently do that, which can be extremely bad if you have multiple templates with the same name.  (The same case where it's bad with normal functions, but you get an error there.)

To be concrete, I am referring to this bit of undefined behavior, which goes away and does what you expect if you put the function templates in anonymous namespaces.
```
$ cat foo.cpp
#include <iostream>
template <typename T>
void f() { std::cout << "foo\n"; }
void foo() { f<int>(); }
$ cat bar.cpp
#include <iostream>
template <typename T>
void f() { std::cout << "bar\n"; }
void bar() { f<int>(); }
$ cat main.cpp
void foo();
void bar();

int main() {
  foo();
  bar();
  return 0;
}
$ g++ foo.cpp bar.cpp main.cpp 
$ ./a.out 
foo
foo
```

Reply:
Ah, I see how you interpreted my statement and what your argument is.